pkg/cache/cache.go

```diff
@@ -201,9 +210,6 @@ type Options struct {
 	// ByObject restricts the cache's ListWatch to the desired fields per GVK at the specified object.
 	// object, this will fall through to Default* settings.
 	ByObject map[client.Object]ByObject
-
-	// newInformer allows overriding of NewSharedIndexInformer for testing.
-	newInformer *func(toolscache.ListerWatcher, runtime.Object, time.Duration, toolscache.Indexers) toolscache.SharedIndexInformer
```

Comment:
Not sure super why this is originally a `*func`, as regular `func`s can already be nil. As our previous fork (kcp-1.26) already introduced `NewInformerFunc client.NewInformerFunc`, I removed this one here and adjusted the related code to not expect a function pointer.

Reply:
is this different func type than `client.NewInformerFunc`?